### PR TITLE
Fix bug: panicked while add single song from search to empty playlist

### PR DESCRIPTION
**Pull request by konstcode**
Panic output:
panicked at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/vec/mod.rs:1562:21:                                                                             
 insertion index (is 1) should be <= len (is 0)

### CODE REVIEW

File: src/structures/sound_action.rs [hunk 1]
```diff
@@ -118,7 +118,10 @@ impl SoundAction {
                         MusicDownloadStatus::NotDownloaded
                     },
                 );
-                player.list.insert(player.current + 1, video);
+                match player.list.len() {
+                    0 => player.list.insert(0, video),
```
Comment (ccgauche) [inline src/structures/sound_action.rs:122]:
```suggestion
                    0 => player.list.push(video),
```

File: src/structures/sound_action.rs [hunk 2]
```diff
@@ -118,7 +118,10 @@ impl SoundAction {
                         MusicDownloadStatus::NotDownloaded
                     },
                 );
-                player.list.insert(player.current + 1, video);
+                match player.list.len() {
```
Comment (ccgauche) [inline src/structures/sound_action.rs:121]: You can replace that with an if .is_empty()

**Review by ccgauche — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution! Here are some little improvements that can be added

File: src/structures/sound_action.rs (hunk 2 above)
Comment (konstcode) [reply src/structures/sound_action.rs]: Yes, thanks I think it would be more wise because of condition simplicity.

**Review by ccgauche — APPROVED** [review]

**Comment by ccgauche** [comment]
Thank you a lot for the contribution. Always appreciate help on the project.

**Pull request merged**